### DISCUSSION TITLE
Troubleshooting "invalid_request" in refresh token flow

### DISCUSSION THREAD

**Discussion by simpleway** [ory/hydra#3442] [Q&A]
A client renew access token through /oauth2/token endpoint daily via nightly batch, but got sporadic errors:

Request Headers:
Accept: application/json
Authorization: ******
Request Content:
grant_type=refresh_token&refresh_token=*******

got following response:
`{
   "error": "invalid_request",
   "error_description": "The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. Make sure that the various parameters are correct, be aware of case sensitivity and trim your parameters. Make sure that the client you are using has exactly whitelisted the redirect_uri you specified."
}`
 
There were many successful token refreshed in same batch with same formulated request (same client_id/secret, but different refresh_token from different users). Since tokens (include refresh tokens) are encrypted at rest in the database, how can I troubleshoot the root cause of this type of sporadic error?

**Comment by riksnelders** [comment]
Not sure if its the same issue but as of late we had a load of errors on the /token endoint using refreshtokens.
We have not updated any of the packages that are using hydra, or hydra itself.

Found a solution yet ?

Edit: logs show errors starting from 2023-04-10 20:36:25.154 CEST
